flake.nix

```diff
@@ -0,0 +1,214 @@
+{
+  nixConfig.bash-prompt = "[nix-develop:]";
+
+  description = "🥝 Next generation of the Kiwi IRC web client";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs?ref=nixos-21.05";
+  inputs.webircgateway = {
+    url = "github:kiwiirc/webircgateway";
+    flake = false;
+  };
+  inputs.kiwiirc-desktop = {
+    url = "github:kiwiirc/kiwiirc-desktop";
+    flake = false;
+  };
+
+  outputs = { self, nixpkgs, webircgateway, kiwiirc-desktop }:
+    let
+      # System types to support.
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
+
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; overlays = [ self.overlay ]; });
+    in {
+      overlay = final: prev:
+        with final; {
+          kiwiirc = mkYarnPackage rec {
```

Comment:
I don't think that it's true that using rec in our context is an antipattern. It can be in more complex scenarios, but this is not a complex scenario. It is common to see rec used in this way. The obvious benefit of keeping it this way is that it's easier to read and doesn't require knowledge of nix.dev's opinions to understand the context of why this harder-to-read code is present.

Reply:
More specifically, I mean to say that the code we've written to use `rec`, is an appropriate usage of `rec` for these two reasons:
1. There are no easy infinite recursion pitfalls in our usage
2. Our code is unlikely to be overriden with overrideAttrs

I could be wrong about 2.

Reply:
I do think 2 is pretty important for packages in general. Seems like a trivial change for a fair amount of benefit IMO

Reply:
Actually, I'm not sure of the behavior on overrideattrs. If it only breaks overriding the recursive attributes, then it doesn't matter since pname/name isn't going to change anyway.